tests/test_unit_proxy.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2018 Snowflake Computing, Inc. All rights reserved.
+ */
+
+#include "util/Proxy.hpp"
+#include "utils/test_setup.h"
+#include "utils/TestSetup.hpp"
+
+typedef ::Snowflake::Client::Util::Proxy Proxy;
+
+void test_proxy_parts_equality(const char *proxy_str, const char *user, const char *pwd, const char *machine, unsigned port, const char *scheme) {
+    Proxy proxy(proxy_str);
+
+    assert_string_equal(proxy.getUser().c_str(), user);
+    assert_string_equal(proxy.getPwd().c_str(), pwd);
+    assert_string_equal(proxy.getMachine().c_str(), machine);
+    assert_int_equal(proxy.getPort(), port);
+    assert_string_equal(proxy.getScheme().c_str(), scheme);
+}
+
+void test_proxy_machine_only(void **unused)
+{
+    test_proxy_parts_equality("localhost", "", "", "localhost", 0, "");
```

Comment:
would this be port 80?

Reply:
I set it to 0 (which means all ports) if the port is not specified and don't set any value in the AWS SDK for proxy port. I don't want to assume a particular port to connect to a proxy server incase 80 is not the default.